test/fakeintake/aggregator/common_test.go

```diff
@@ -127,11 +129,34 @@ func TestCommonAggregator(t *testing.T) {
 	})
 
 	t.Run("Reset", func(t *testing.T) {
-		_, err := generateTestData()
+		data, err := generateTestData()
 		require.NoError(t, err)
 		agg := newAggregator(parseMockPayloadItem)
+		err = agg.UnmarshallPayloads(data)
+		require.NoError(t, err)
+		assert.NotEmpty(t, agg.payloadsByName)
 		agg.Reset()
-		assert.Equal(t, 0, len(agg.payloadsByName))
-		validateCollectionTime(t, agg)
+		assert.Empty(t, agg.payloadsByName)
+	})
+
+	t.Run("Thread safe", func(t *testing.T) {
```

Comment:
I like this test to test for race conditions. However, isn'it a test that would be flaky by design? Or can we be sure that it will always fail if a race condition exists and always succeed otherwise?

Reply:
I don't know if it's the `-race` flag, could not find if it helps flagging race condition when forced in tests too. What this test does is accessing the set and the get from two separate go routines, and we expect to never fails.

We first ensure that the aggregator has data (otherwise yes it would be flaky by design).

Then we reset to the same values 100 times, and at the same time we get values. As get and set operations are atomic, and as they are synced by the same mutex, if the code works properly this never fails, thanks to the lock.

Reply:
Well spotted I was not waiting for those routines to complete 🙈 . Fixed !